### DISCUSSION TITLE
Use sr25519 on Web3Auth

### DISCUSSION THREAD

**Discussion by ThibTcorplabs** [Web3Auth/Web3Auth#1017] [Help]
I am a dev working on a blockchain using sr25519, is it possible to use web3auth ?

**Comment by Saviour1001** [comment]
Yes.
You will require to change the schema of the key. For example, [here](https://github.com/Web3Auth/examples/blob/50b04b8a4343c3de8487114b0b671ec939b9aef2/web-modal-sdk/multi-chain/react-multi-chain-web3auth-example/src/App.tsx#L213) we are shifting to Ed25519 using some helper libraries. 

```
 const privateKey = await rpc.getPrivateKey();
 const { getED25519Key } = await import("@toruslabs/openlogin-ed25519");
 const ed25519key = getED25519Key(privateKey).sk.toString("hex");
```

**Comment by ThibTcorplabs** [comment]
Thanks a lot for your answer @Saviour1001 ! 
It appears that, according to the documentation, Web3Auth "supports all blockchains following the secp256k1 & ed25519 curves". So yes you can switch to ED25519 but I am not sure about SR25519 ?

**Comment by Saviour1001** [comment]
We added a [Polkadot](https://github.com/Web3Auth/examples/tree/main/web-modal-sdk/polkadot) example !!
